### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by fnattino**
Check that docs are up-to-date, add API section (fixing docstrings):
- [x] co-clustering
- [x] tri-clustering
- [x] k-means
- [x] utils

### CODE REVIEW

File: cgc/utils.py [hunk 1]
```diff
@@ -155,24 +178,30 @@ def calculate_tricluster_averages(Z, row_clusters, col_clusters, bnd_clusters,
 def calculate_cluster_feature(Z, function, clusters, nclusters=None, **kwargs):
     """
     Calculate feature for clusters. This function works in N dimensions
```
Comment (rogerkuou) [inline cgc/utils.py:180]:
```suggestion
    Calculate features for clusters. This function works in N dimensions
```

File: docs/coclustering.rst [hunk 2]
```diff
@@ -0,0 +1,147 @@
+Co-clustering
+=============
+
+Introduction
+------------
+
+The ``coclustering`` module provides the functionality to perform the co-clustering analysis of a positive data matrix
+with real-valued elements. The code implements the Bregman block-average co-clustering (BBAC) algorithm from Ref. [#]_
+and it was inspired from the Matlab `code`_ by Srujana Merugu and Arindam Banerjee.
```
Comment (rogerkuou) [inline docs/coclustering.rst:9]:
```suggestion
and it was inspired by the Matlab `code`_ by Srujana Merugu and Arindam Banerjee.
```

File: docs/coclustering.rst [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    )
+
+Here, we have set the maximum number of row and column clusters to 2 and 3, respectively. However, a lower number of
+clusters can be identified by the algorithm (some of the clusters may remain empty). The algorithm entails an iterative
+procedure that is considered converged when the error of two consecutive iterations differs by less than a threshold
+(the default value is 1.e-5).
+
+Multiple runs should be performed in order to limit the influence of the choice of initial cluster assignment on the
+result. A numerical parameter guarantees that no zero-valued arguments are encountered in the logarithm that appears in
+the I-divergence expression, which is employed as objective function. Results are (optionally) written to a JSON file.
```
Comment (rogerkuou) [inline docs/coclustering.rst:55]:
```suggestion
the I-divergence expression, which is employed as an objective function. Results are (optionally) written to a JSON file.
```

File: docs/coclustering.rst [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+architectures with multi-core CPUs, the various differently-initialized co-clustering runs can be executed as multiple
+threads. They are, in fact, embarrassingly parallel tasks that require no communication between each other. The
+co-clustering analysis is run using e.g. 4 threads as:
+
+.. code-block:: python
+
+    results = cc.run_with_threads(nthreads=4)
+
+This first implementation makes use of (fast) matrix multiplications to calculate cluster-based properties, such as
+averages and distances. However, if ``Z``'s dimensions are large, large auxiliary matrices needs to be stored into
```
Comment (rogerkuou) [inline docs/coclustering.rst:73]:
```suggestion
averages and distances. However, if ``Z``'s dimensions are large, large auxiliary matrices need to be stored into
```

File: docs/coclustering.rst [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+In a second Dask-based implementation, the various co-clustering runs are submitted to the Dask scheduler, which
+distributes them across the cluster. This implementation, which is activated by setting ``low_memory=False``, is
+experimental and it typically leads to very large memory usages.
+
+Results
+-------
+
+The ``CoclusteringResults`` object returned by ``Coclustering.run_with_threads`` and ``Coclustering.run_with_dask``
+contains the final row and column cluster assigments (``results.row_clusters`` and ``results.col_clusters``,
```
Comment (rogerkuou) [inline docs/coclustering.rst:129]:
```suggestion
contains the final row and column cluster assignments (``results.row_clusters`` and ``results.col_clusters``,
```

File: docs/kmeans.rst [hunk 6]
```diff
@@ -0,0 +1,90 @@
+K-means refinement
+==================
+
+Introduction
+------------
+
+The `Kmeans` module is an implementation of the `k-means clustering`_ to refine the results of a co-clustering or
+tri-clustering calculation. This k-mean refinement allows to identify similarity patterns between co- or tri-clusters.
```
Comment (rogerkuou) [inline docs/kmeans.rst:8]:
```suggestion
tri-clustering calculation. This k-mean refinement allows identifying similarity patterns between co- or tri-clusters.
```

File: docs/kmeans.rst [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+#. Mean value;
+#. Standard deviation;
+#. Minimum value;
+#. Maximum value;
+#. 5th percentile;
+#. 95th percentile;
+
+The implementation, which is based on the `scikit-learn`_ package, tests a range of k values and select the optimal one
+on the basis of the `Silhouette coefficient`_.
```
Comment (rogerkuou) [inline docs/kmeans.rst:20]:
```suggestion
based on the `Silhouette coefficient`_.
```

File: docs/triclustering.rst [hunk 8]
```diff
@@ -0,0 +1,115 @@
+Tri-clustering
+==============
+
+Introduction
+------------
+
+The ``triclustering`` module provides a generalization of the co-clustering algorithm to three-dimensional arrays (see
+Ref. [#]_). For geospatial data, tri-clustering analyses allow to extend the search for similarity patterns in
```
Comment (rogerkuou) [inline docs/triclustering.rst:8]:
```suggestion
Ref. [#]_). For geospatial data, tri-clustering analyses allow extending the search for similarity patterns in
```

File: docs/triclustering.rst [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+.. code-block:: python
+
+    results = tc.run_with_threads(nthreads=4)
+
+Distributed (Dask-based)
+************************
+
+Also for the tri-clustering, analysis on distributed systems can be carried out using Dask (see also
+:doc:`coclustering`). Once connection to a Dask cluster is setup:
```
Comment (rogerkuou) [inline docs/triclustering.rst:74]:
```suggestion
:doc:`coclustering`). Once the connection to a Dask cluster is setup:
```

File: docs/triclustering.rst [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+.. code-block:: python
+
+    results = tc.run_with_dask(client)
+
+Results
+-------
+
+The ``TriclusteringResults`` object returned by ``Triclustering.run_with_threads`` and ``Triclustering.run_with_dask``
+contains the final row, column, and band cluster assigments (``results.row_clusters``, ``results.col_clusters``, and
```
Comment (rogerkuou) [inline docs/triclustering.rst:93]:
```suggestion
contains the final row, column, and band cluster assignments (``results.row_clusters``, ``results.col_clusters``, and
```

**Review by rogerkuou — APPROVED** [review]
Hi @fnattino , I think this is really awesome! Just added some picky comments based on the help from Grammarly. Please feel free to merge it!

**Comment by fnattino** [comment]
Thanks so much @rogerkuou !

**Pull request merged**